### ISSUE TITLE
re-ordering within facet based on levels within a factor

### ISSUE THREAD

**Issue by KMKemp** [joey711/phyloseq#734]
I have created a faceted stacked bar chart and would like to reorder the samples within each facet based on metadata associated with the samples. 

The chart is faceted by this factor, which has four levels: H, 24h.PX, 3.5d.PX, 8.10d.PX
>factor(sample_data(LKTS_2015_R2_Order)$Lesion_age)
...
Levels: H 24h.PX 3.5d.PX 8.10d.PX

Now that the data is faceted correctly, I would like to reorder the samples within each facet according to this factor, which has four levels: H, front, 2.cm, 10.cm. I have ordered the levels in the factor by:

>sample_data(LKTS2015_R2)$Distance <- factor(sample_data(LKTS2015_R2)$Distance, levels = c("H", "front", "2.cm", "10.cm"))

>factor(sample_data(LKTS_2015_R2_Order)$Distance)
...
Levels: H front 2.cm 10.cm

However, the samples within each facet are not reordering.

Here is my bar chart code:
ggplot(LKTS_2015_R2_Order, aes(x = Sample, y = Abundance, fill = Order)) + geom_bar(stat = "identity", colour = "black")  + scale_fill_manual(values = K13rainbow) + labs(x = " ",y = "Relative Abundance (%)") + facet_grid(~Lesion_age, space = "free", scales="free") + theme(axis.text.x = element_text(angle = 90, hjust = 1))

**Comment by KMKemp** [comment]
I am responding with my solution, because it seems as those multiple people have asked this question. After working on this issue for some time ... I opted to create a new sample variable and added letters to the beginning of the values in the way I wanted samples to be ordered on the x-axis (e.g., "A_", "B_", ...). Then I plotted samples on the x-axis by this new variable and this did what I wanted. I will now go back and add manual x-labels using ggplot. This is the best solution I could come up with. 

I am providing an example of the plot below.

![rplot05](https://cloud.githubusercontent.com/assets/20914182/24324741/aab81750-1159-11e7-8a86-fca07c8632f3.png)
scale_x_discrete(labels=c())
